graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,273 @@
+            raise ValueError(msg)
+
+        seedsA = y[0]
+        seedsB = y[1]
+
+        if len(seedsA) != len(seedsB):
+            msg = "Must have the same number of seeds for each adjacency matrix"
+            raise ValueError(msg)
+        if len(seedsA) == 0:
+            print("Must include at least one seed to produce nomination list")
+            return None
+
+        voi = np.reshape(np.array(voi), (1,))
+
+        # get reordering for A
+        nsx1 = np.setdiff1d(np.arange(self.G_1.shape[0]), np.concatenate((seedsA, voi)))
+        a_reord = np.concatenate((seedsA, voi, nsx1))
+
+        # get reordering for B
+        nsx2 = np.setdiff1d(np.arange(self.G_2.shape[0]), seedsB)
+        b_reord = np.concatenate((seedsB, nsx2))
+
+        AA = self.G_1[np.ix_(a_reord, a_reord)]
+        BB = self.G_2[np.ix_(b_reord, b_reord)]
+
+        seeds_reord = np.arange(len(seedsA))
+        voi_reord = len(seedsA)
+
+        subgraph_AA = np.array(_ego_list(AA, self.h, voi_reord, mindist=1))
+
+        voi_reord = np.reshape(np.array(voi_reord), (1,))
+
+        Sx1 = np.intersect1d(a_reord[subgraph_AA], seeds_reord)
+        Sx2 = np.intersect1d(np.arange(BB.shape[0]), Sx1)
+
+        if len(Sx2) <= 0:
+            print("Voi was not a member of the induced subgraph A[seedsA]")
+            return None
+
+        Nx1 = np.array(_ego_list(AA, self.ell, list(Sx1), mindist=0))
+        Nx2 = np.array(_ego_list(BB, self.ell, list(Sx2), mindist=0))
+
+        foo = np.concatenate((Sx1, voi_reord))
+        ind1 = np.concatenate((Sx1, voi_reord, np.setdiff1d(Nx1, foo)))
+        ind2 = np.concatenate((Sx2, np.setdiff1d(Nx2, Sx2)))
+
+        AA_fin = AA[np.ix_(ind1, ind1)]
+        BB_fin = BB[np.ix_(ind2, ind2)]
+
+        seeds_fin = list(range(len(Sx1)))
+        sgm = GMP(
+            n_init=self.R, shuffle_input=False, init_method="rand", padding="naive"
+        )
+        corr = sgm.fit_predict(AA_fin, BB_fin, seeds_A=seeds_fin, seeds_B=seeds_fin)
+        P_outp = sgm.probability_matrix_
+
+        b_inds = b_reord[ind2]
+        self.n_seeds_used = len(Sx1)
+
+        nomination_list = list(zip(b_inds[self.n_seeds_used :], P_outp[0]))
+        nomination_list.sort(key=lambda x: x[1], reverse=True)
+        self.nomination_list = nomination_list
+        return self
+
+    def fit_predict(self, X, y=[]):
+        """
+        Fits model to two adjacenty matricies and returns nomination list
+
+        Parameters
+        ----------
+        X: int
+            Vertex of interest (voi)
+
+        y: list
+            List of length two `[seedsA, seedsB]` where first element is
+        the seeds associated with adjacency matrix G_1
+        and the second element the adjacency matrix associated with G_2, note
+        `len(seedsA)==len(seedsB)` The elements of `seeds_A` and `seeds_B` are
+        vertices which are known to be matched, that is, `seeds_A[i]` is matched
+        to vertex `seeds_B[i]`.
```

Comment:
This description and others formatted like it are displaying improperly in docs, they are highlighted grey.